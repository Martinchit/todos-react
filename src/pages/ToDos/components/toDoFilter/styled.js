import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { IoCheckmarkDoneCircleSharp, IoAlertCircleSharp } from 'react-icons/io5';
import { Box } from '../../../../core/components/atoms/Box';

export const ToDoFilterContainer = styled(Box)`
  box-shadow: ${({ theme }) => `0 -5px 5px -7px ${theme.colors.black}`};
`;

export const ToDoStatsDoneIcon = styled(IoCheckmarkDoneCircleSharp)`
  color: ${({ theme }) => theme.colors.success};
  font-size: 1.5rem;
  margin-right: 3px;
`;

export const ToDoStatsActiveIcon = styled(IoAlertCircleSharp)`
  color: ${({ theme }) => theme.colors.warning};
  font-size: 1.5rem;
  margin-right: 3px;
`;

export const ToDoFilterOption = styled(Link)`
  margin: 0 3px;
  border-bottom: ${({ $selected, theme }) =>
    $selected ? `1px solid ${theme.colors.primary}` : 'none'};
  font-weight: ${({ $selected }) => ($selected ? 800 : 400)};
  color: ${({ theme }) => theme.colors.black};
  text-decoration: none;
  padding: 0 10px;
  cursor: pointer;
`;

export const CompletedToDoDeleteButton = styled.button`
  color: ${({ theme }) => theme.colors.danger};
  cursor: pointer;
  font-weight: 600;
`;
