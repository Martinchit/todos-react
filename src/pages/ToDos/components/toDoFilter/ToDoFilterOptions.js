import React from 'react';
import PropTypes from 'prop-types';
import { ToDoFilterOption } from './styled';
import { Box } from '../../../../core/components/atoms/Box';
import { TO_DO_STATUS } from '../../../../core/lib/utils/toDoStatus';

export const ToDoFilterOptions = ({ currentFilterOption }) => {
  const filters = React.useMemo(
    () => [
      {
        name: 'All',
        link: '/todos',
      },
      {
        name: 'Active',
        link: '/todos/active',
      },
      {
        name: 'Completed',
        link: '/todos/completed',
      },
    ],
    []
  );
  return (
    <Box width="40%" justifyContent="space-between">
      {filters.map(({ name, link }) => (
        <ToDoFilterOption
          key={name}
          to={link}
          $selected={currentFilterOption === name.toLowerCase()}
        >
          {name}
        </ToDoFilterOption>
      ))}
    </Box>
  );
};

ToDoFilterOptions.propTypes = {
  currentFilterOption: PropTypes.oneOf(Object.values(TO_DO_STATUS)),
};

ToDoFilterOptions.defaultProps = {
  currentFilterOption: TO_DO_STATUS.ALL,
};
