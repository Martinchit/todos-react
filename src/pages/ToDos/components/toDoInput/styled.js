import styled from 'styled-components';
import { TiDeleteOutline, TiArrowSortedUp, TiArrowSortedDown } from 'react-icons/ti';

export const ToDoInputField = styled.input`
  width: 80%;
  padding: 2%;
  background-color: transparent;

  :focus,
  :active,
  :hover {
    outline: none;
  }

  ::placeholder,
  ::-webkit-input-placeholder {
    font-style: italic;
  }
  :-ms-input-placeholder {
    font-style: italic;
  }
`;

export const DeleteIcon = styled(TiDeleteOutline)`
  font-size: 1.5rem;
  margin: 0 3%;
  cursor: pointer;
  visibility: ${({ $visible }) => ($visible ? 'visible' : 'hidden')};
  color: ${({ theme }) => theme.colors.danger};
`;

export const ExpandIcon = styled(TiArrowSortedDown)`
  font-size: 1.5rem;
  margin: 0 3%;
  cursor: pointer;
  visibility: ${({ $visible }) => ($visible ? 'visible' : 'hidden')};
  color: ${({ theme }) => theme.colors.primary};
`;

export const HideIcon = styled(TiArrowSortedUp)`
  font-size: 1.5rem;
  margin: 0 3%;
  cursor: pointer;
  visibility: ${({ $visible }) => ($visible ? 'visible' : 'hidden')};
  color: ${({ theme }) => theme.colors.black};
`;
