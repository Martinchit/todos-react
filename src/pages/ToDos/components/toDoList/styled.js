import styled from 'styled-components';
import { RiDeleteBinLine } from 'react-icons/ri';
import { Box } from '../../../../core/components/atoms/Box';

export const ToDoItemContainer = styled(Box)`
  box-shadow: ${({ theme }) => `0 -5px 5px -7px ${theme.colors.black}`};
`;

export const ToDoLabel = styled.label`
  width: 80%;
  cursor: pointer;
`;

export const ToDoEditInput = styled.input`
  height: 100%;
  width: 86%;
  background-color: transparent;

  :focus,
  :active,
  :hover {
    outline: none;
  }

  :focus {
    border-bottom: ${({ theme }) => `2px solid ${theme.colors.success}`};
  }
`;

export const ItemDeleteButton = styled(RiDeleteBinLine)`
  font-size: 1.5rem;
  cursor: pointer;
  color: ${({ theme }) => theme.colors.danger};
`;
