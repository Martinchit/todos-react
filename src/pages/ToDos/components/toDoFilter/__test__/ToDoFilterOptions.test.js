import React from "react";
import { shallow } from 'enzyme'; 
import 'jest-styled-components';

import { ToDoFilterOptions } from "../ToDoFilterOptions";
import { ToDoFilterOption } from '../styled';

describe("ToDoFilterOptions", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ToDoFilterOptions />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(ToDoFilterOption).length).toBe(3);
  })
})