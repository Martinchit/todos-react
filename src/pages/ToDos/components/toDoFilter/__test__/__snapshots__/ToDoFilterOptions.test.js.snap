// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoFilterOptions renders correctly 1`] = `ShallowWrapper {}`;
